feat(android): Migration docs for 6.3.0 (#5298)
getsentry/sentry-docs: src/platforms/android/migration.mdx

description: "Migrating between versions of Sentry's SDK for Android."
---

## Migrating from `io.sentry:sentry-android 6.2.x` to `io.sentry:sentry-android 6.3.0`

> * `sentry-android-okhttp`, `sentry-android-fragment` and `sentry-android-timber` packages now declare their upstream dependencies as `compileOnly`. Please make sure to have `okhttp`, `fragment` and `timber` dependencies on the classpath if you are using those integrations.

Reviewer: `m`: Can we maybe add what happens if you don't have them on the classpath?

Author: Added that the app will crash in that case 👍

> * `sentry-android-okhttp`, `sentry-android-fragment` and `sentry-android-timber` packages now declare their upstream dependencies as `compileOnly`. Please make sure to have `okhttp`, `fragment` and `timber` dependencies on the classpath if you are using those integrations.

Author: ```suggestion
* `sentry-android-okhttp`, `sentry-android-fragment` and `sentry-android-timber` packages now declare their upstream dependencies as `compileOnly`. Please make sure to have `okhttp`, `fragment` and `timber` dependencies on the classpath if you are using those integrations, otherwise, the application will crash with a `NoClassDefFoundError` exception.
```

> * `sentry-android-okhttp`, `sentry-android-fragment` and `sentry-android-timber` packages now declare their upstream dependencies as `compileOnly`. Please make sure to have `okhttp`, `fragment` and `timber` dependencies on the classpath if you are using those integrations, otherwise, the application will crash with a `NoClassDefFoundError` exception.

Reviewer 2: Let's avoid starting the sentence with code format and we shouldn't use bullets when there's only going to be one bullet point. Also wondering if we can add the word "integration" before "packages" to make clearer that those are the integrations we're talking about in the second sentence.

```suggestion
The `sentry-android-okhttp`, `sentry-android-fragment`, and `sentry-android-timber` integration packages now declare their upstream dependencies as `compileOnly`. Make sure to have the `okhttp`, `fragment`, and `timber` dependencies on the classpath if you're using those integrations or the application will crash with a `NoClassDefFoundError` exception.
```

Author: Thanks, we can just use "integrations" there and drop the "packages" altogether
